app/celery/research_mode_tasks.py

```diff
@@ -50,42 +43,16 @@ def send_email_response(reference, to):
     process_ses_results.apply_async([body], queue=QueueNames.RESEARCH_MODE)
 
 
-def make_request(notification_type, provider, data, headers):
-    api_call = "{}/notifications/{}/{}".format(current_app.config["API_HOST_NAME"], notification_type, provider)
-
-    try:
-        response = request("POST", api_call, headers=headers, data=data, timeout=60)
-        response.raise_for_status()
-    except RequestException as e:
-        api_error = HTTPError(e)
-        # TODO: Revert the following log to error level once the SMS research simulation has been fixed via:
-        #       https://trello.com/c/FTzY1HLJ/563-rework-fake-sms-callbacks-for-notifications-sent-using-test-keys
-        current_app.logger.info("API {} request on {} failed with {}".format("POST", api_call, api_error.response))
-        # TODO: Reinstate the error raise once the SMS research simulation has been fixed via:
-        #       https://trello.com/c/FTzY1HLJ/563-rework-fake-sms-callbacks-for-notifications-sent-using-test-keys
-        # raise api_error
-    finally:
-        current_app.logger.info("Mocked provider callback request finished")
-    return response.json()
-
+def aws_ses_callback(notification_id, to):
+    now = datetime.now()
+    timestamp = now.strftime("%m/%d/%Y %H:%M:%S")
 
-def firetext_callback(notification_id, to):
-    """
-    status: 0 - delivered
-    status: 1 - perm failure
-    """
     if to.strip().endswith(perm_fail):
-        status = "1"
+        return sns_failed_callback("Permanent failure", notification_id, destination=to, timestamp=timestamp)
     elif to.strip().endswith(temp_fail):
-        status = "2"
+        return sns_failed_callback("Temporary failure", notification_id, destination=to, timestamp=timestamp)
     else:
-        status = "0"
-    return {
-        "mobile": to,
-        "status": status,
-        "time": "2016-03-10 14:17:00",
-        "reference": notification_id,
-    }
+        return sns_success_callback(notification_id, destination=to, timestamp=timestamp)
```

Comment:
Your PR description says that you check only the last digit but it doesn't seem to work this way? That's fine, we should stick with numbers defined. Do we need to update the documentation? https://documentation.notification.canada.ca/en/keys.html#test

Reply:
The defined numbers feature in code and documentation are broken. I tried with the numbers as-is and with a `+` sign in front. It does not pass input validation for the numbers we expect. Maybe there is something more one would need to do, but then the documentation does not mention it. 

I didn't push the code in there yet, my bad. My work day yesterday was interrupted with an electricity outage while I was working on polishing this PR.